### PR TITLE
Feat/mpp 187 in page

### PR DESCRIPTION
**Pull request by joyet-simon**

### CODE REVIEW

**Review by ClaireAlma — APPROVED** [review]

File: cartridges/int_alma/cartridge/scripts/helpers/almaCheckoutHelper.js [hunk 1]
```diff
@@ -149,20 +149,20 @@ function getPropertiesForPlan(plan, currencyCode) {
 /**
  * Returns if the current payment option is pnx 2,3 or 4
  * @param  {Object} plan any alma plan
- * @returns {boolean} true means we can use fragment
+ * @returns {boolean} true means we can use inpage
  */
 function isPnx(plan) {
-    return plan.installments_count <= 4;
+    return plan.installments_count <= 4 && plan.deferred_days <= 0;
```
Comment (Benjamin-Freoua-Alma) [inline cartridges/int_alma/cartridge/scripts/helpers/almaCheckoutHelper.js:155]: isPnx function is now with Pay Now, Do you not separate PnX and Pay Now ?

File: cartridges/int_alma/cartridge/scripts/helpers/almaPaymentHelper.js [hunk 2]
```diff
@@ -297,7 +297,7 @@ function buildPaymentData(installmentsCount, deferredDays, locale) {
             ipn_callback_url: URLUtils.http('Alma-IPN').toString(),
             customer_cancel_url: URLUtils.https('Alma-CustomerCancel').toString(),
             locale: locale,
-            origin: 'online',
+            origin: 'online_in_page',
```
Comment (Benjamin-Freoua-Alma) [inline cartridges/int_alma/cartridge/scripts/helpers/almaPaymentHelper.js:300]: Is well only for InPage ? There are not condition to send 'online' if is not InPage ?

**Review by Benjamin-Freoua-Alma — APPROVED** [review]

**Review by Francois-Gomis — APPROVED** [review]

File: cartridges/int_alma/cartridge/scripts/helpers/almaPaymentHelper.js [hunk 3]
```diff
@@ -280,11 +280,17 @@ function buildPaymentData(installmentsCount, deferredDays, locale) {
     var BasketMgr = require('dw/order/BasketMgr');
     var URLUtils = require('dw/web/URLUtils');
     var almaHelper = require('*/cartridge/scripts/helpers/almaHelpers');
+    var almaCheckoutHelper = require('*/cartridge/scripts/helpers/almaCheckoutHelper');
 
     var formatAddress = require('*/cartridge/scripts/helpers/almaAddressHelper').formatAddress;
     var isOnShipmentPaymentEnabled = require('*/cartridge/scripts/helpers/almaOnShipmentHelper').isOnShipmentPaymentEnabled;
     var formatCustomerData = require('*/cartridge/scripts/helpers/almaHelpers').formatCustomerData;
 
+    var online = 'online';
```
Comment (Francois-Gomis) [inline cartridges/int_alma/cartridge/scripts/helpers/almaPaymentHelper.js:289]: I think var name can be origin  ?

**Review by Francois-Gomis — APPROVED** [review]

File: cartridges/int_alma/cartridge/scripts/helpers/almaCheckoutHelper.js (hunk 1 above)
Comment (Francois-Gomis) [reply cartridges/int_alma/cartridge/scripts/helpers/almaCheckoutHelper.js]: I think it's function name is not good

Comment (joyet-simon) [reply cartridges/int_alma/cartridge/scripts/helpers/almaCheckoutHelper.js]: Done

**Review by Francois-Gomis — APPROVED** [review]

**Pull request merged**